In [1]:
import pandas as pd
import matplotlib.pyplot as plt

# Leukemia Diagnosis

In [2]:
# load the data
url = 'https://raw.githubusercontent.com/um-perez-alvaro/Data-Science-Practice/master/Data/leukemia.csv'
data = pd.read_csv(url)
data.head()

0      1      2     3      4      5     6      7     8      9  ...  \
0 -73.0  -69.0  -48.0  13.0  -86.0 -147.0 -65.0  -71.0 -32.0  100.0  ...   
1 -16.0  -63.0  -97.0 -42.0  -91.0 -164.0 -53.0  -77.0 -17.0  122.0  ...   
2   4.0  -45.0 -112.0 -25.0  -85.0 -127.0  56.0 -110.0  81.0   41.0  ...   
3 -31.0 -110.0  -20.0 -50.0 -115.0 -113.0 -17.0  -40.0 -17.0   80.0  ...   
4 -33.0  -39.0  -45.0  14.0  -56.0 -106.0  73.0  -34.0  18.0   64.0  ...   

   16054  16055  16056  16057  16058  16059  16060  16061  16062  label  
0  352.0  -67.0  121.0   -5.0  -11.0  -21.0  -41.0 -967.0 -120.0      1  
1  244.0  -15.0  119.0  -32.0    4.0  -14.0  -28.0 -205.0  -31.0      1  
2  163.0  -14.0    7.0   15.0   -8.0 -104.0  -36.0 -245.0   34.0      1  
3  625.0   18.0   59.0  -10.0   32.0   -2.0   10.0 -495.0  -37.0      1  
4  398.0   38.0  215.0   -2.0   44.0    3.0   68.0 -293.0  -34.0      1  

[5 rows x 16064 columns]

In [16]:
data.shape

(198, 16064)

This dataset contains cancer gene-expression levels (16,063 genes, 198 samples) from *Multiclass cancer diagnosis using tumor gene expression signatures (Golub et all, 2001)*

Cancer classes are labelled as follows:


- 1: breast
- 2:  prostate
- 3:  lung
- 4:  collerectal
- 5:  lymphoma
- 6:  bladder
- 7:  melanoma
- 8:  uterus
- 9:  **leukemia**
- 10: renal
- 11: pancreas
- 12: ovary
- 13: meso
- 14: cns


The goal is to train a linear model to classify cancers as either leukemia or not-leukemia

**Part 1:** Replace the label values with:

$$
\left\{ \begin{array}{ll} 1 & \mbox{ if cancer is leukemia}\\
0 & \mbox{ if cancer is not leukemia}. \end{array}\right.
$$

In [4]:
data.label = data.label.apply(lambda x: 1 if x==9 else 0)

0    168
1     30
Name: label, dtype: int64

**Part 2:** Define X and y from the "data" DataFrame, and then split X and y into training and testing sets

**Part 3:** Use the training set to train a **Lasso regression model**. 
How many model coefficients are equal to 0?

**Part 4:**  Use your linear model to classify all the cancers in the test set.  
Use the rule:

$$
\mbox{classify cancer as a }\left\{ \begin{array}{ll} 
\mbox{leukemia} & \mbox{ if y_test_pred}>0.5 \\
\mbox{not leukemia }5 & \mbox{ if predicted y_test_pred}\leq 0.5
\end{array}\right.
$$

How many cancers are misclassified? 